src/main/java/com/lovetropics/minigames/common/core/game/behavior/action/GameActionList.java

```diff
@@ -13,92 +12,82 @@
 import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.util.StringRepresentable;
 
 import java.util.Arrays;
 import java.util.List;
 import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class GameActionList<T, A extends ActionTarget<T>> {
```

Comment:
Not sure if `A` is actually required? Might be a bit nicer in usages to only pass `T`.

Also - we're using the raw type in a lot of places now, which would be nice to avoid.